### PR TITLE
Catch error when unable to send DM

### LINKED ISSUE #797: AutoPruneHelperRoutine crashes on DM when DMs closed
The `AutoPruneHelperRoutine` class crashes when attempting to send a DM to someone who does not accept DMs.

stacktrace

Instead, it should just log the issue on debug level. Therefore, simply add a failure handler `.queue(() -> {}, failure -> logger.debug(..., failure))`.

here

### PR DESCRIPTION
**Pull request by christolis**
Fixes #797. I was unable to test this, but I think this might be a solid solution.

### CODE REVIEW

**Comment by SquidXTV** [comment]
you can maybe instead supply `null` for the success `Consumer`.
not sure if its better/cleaner code then though

> success - The success callback that will be called at a convenient time for the API. (can be null to use default)

https://docs.jda.wiki/net/dv8tion/jda/api/requests/RestAction.html#queue(java.util.function.Consumer,java.util.function.Consumer)

**Comment by surajkumar** [comment]
Maybe we can have the users name in the log message? Though not too important

**Comment by SquidXTV** [comment]
> Maybe we can have the users name in the log message? Though not too important

wont it be in the exception?

**Review by SquidXTV — APPROVED** [review]

**Review by Taz03 — DISMISSED** [review]

File: application/src/main/java/org/togetherjava/tjbot/features/help/AutoPruneHelperRoutine.java [hunk 1]
```diff
@@ -163,7 +163,8 @@ private void pruneMemberFromRole(Member member, Role role, ForumChannel helpForu
         guild.removeRoleFromMember(member, role)
             .flatMap(any -> member.getUser().openPrivateChannel())
             .flatMap(channel -> channel.sendMessage(dmMessage))
-            .queue();
+            .queue(null,
+                    failure -> logger.debug("Tried to send a DM to a user but failed: ", failure));
```
Comment (Zabuzard) [inline application/src/main/java/org/togetherjava/tjbot/features/help/AutoPruneHelperRoutine.java:167]: the log message is missing context. should add sth like
`Failed sending a DM to user (1234567) while pruning them from a helper role.` (with the number being the user ID)

**Review by Zabuzard — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/features/help/AutoPruneHelperRoutine.java (hunk 1 above)
Comment (christolis) [reply application/src/main/java/org/togetherjava/tjbot/features/help/AutoPruneHelperRoutine.java]: Done! 🙌

File: application/src/main/java/org/togetherjava/tjbot/features/help/AutoPruneHelperRoutine.java [hunk 2]
```diff
@@ -163,7 +163,9 @@ private void pruneMemberFromRole(Member member, Role role, ForumChannel helpForu
         guild.removeRoleFromMember(member, role)
             .flatMap(any -> member.getUser().openPrivateChannel())
             .flatMap(channel -> channel.sendMessage(dmMessage))
-            .queue();
+            .queue(null, failure -> logger
+                .debug("Failed sending a DM to user (%s) while pruning them from a helper role."
+                    .formatted(member.getId()), failure));
```
Comment (Taz03) [inline application/src/main/java/org/togetherjava/tjbot/features/help/AutoPruneHelperRoutine.java:168]: use [logger formatter](https://github.com/apache/logging-log4j2#usage)

**Review by Taz03 — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/features/help/AutoPruneHelperRoutine.java (hunk 2 above)
Comment (christolis) [reply application/src/main/java/org/togetherjava/tjbot/features/help/AutoPruneHelperRoutine.java]: It's now done! ✅

**Review by Taz03 — APPROVED** [review]

**Review by Zabuzard — APPROVED** [review]

**Pull request merged**